components/calendar/src/chinese_based.rs

```diff
@@ -309,6 +337,117 @@ impl<C: ChineseBased + CalendarArithmetic> ChineseBasedDateInner<C> {
         debug_assert!(result < MAX_ITERS_FOR_MONTHS_OF_YEAR, "The given year was not a leap year and an unexpected number of iterations occurred searching for a leap month.");
         result
     }
+
+    /// Create a new arithmetic date from a year, month ordinal, and day with bounds checking; returns the
+    /// result of creating this arithmetic date, as well as a ChineseBasedCache - either the one passed in
+    /// optionally as an argument, or a new ChineseBasedCache for the given year, month, and day args.
+    pub(crate) fn new_from_ordinals(
+        year: i32,
+        month: u8,
+        day: u8,
+        cache: ChineseBasedCache,
```

Comment:
Suggestion: If you're not taking ownership of the `ChineseBasedCache`, this should be a reference. (I don't think we should bind ourselves by having it implement `Copy`)

Reply:
I think we should, I would like AnyDateInner to be Copy eventually.

Though that doesn't actually affect the Copy-ness of ChineseBasedCache (which is a private type): that affects the Copy-ness of ChineseBasedDateInner and ChineseDateInner

Reply:
For now at least I made all `ChineseBasedCache` parameters take references